src/ray/gcs/redis_client.cc

```diff
@@ -25,6 +25,30 @@ namespace ray {
 
 namespace gcs {
 
+static int DoGetNextJobID(redisContext *context) {
+  int num_attempts = 0;
+  redisReply *reply = nullptr;
+  while (num_attempts < RayConfig::instance().redis_db_connect_retries()) {
```

Comment:
It seems that there is a lot of duplicated code in this file. Is it possible to simplify it?

Reply:
ok. function abstracted.